assume/common/units_operator.py

```diff
@@ -327,7 +329,14 @@ def get_actual_dispatch(
                         data[key] = unit.outputs[key][start:end]
 
             data["unit"] = unit_id
-            unit_dispatch_dfs.append(data)
+            return data
+
+        with ThreadPoolExecutor() as executor:
+            results = executor.map(
+                lambda unit_id_unit: process_unit(*unit_id_unit), self.units.items()
+            )
+            unit_dispatch_dfs.extend(results)
```

Comment:
are you sure the ThreadPoolExecuter helps here?
From what I know, you probably need to use a `multiprocessing.Pool` to do this parallel..?